src/qiboml/models/encoding.py

```diff
@@ -44,8 +46,9 @@ def __post_init__(
             self._circuit.add(gates.RY(q, theta=0.0, trainable=False))
 
     def _set_phases(self, x: ndarray):
-        for gate, phase in zip(self._circuit.parametrized_gates, x.ravel()):
-            gate.parameters = phase
+        phase = tf.reshape(x, [-1])
```

Comment:
we cannot have `tf` functions here, this has to work for `pytorch` as well and eventually `jax` when we introduce the `Flax` interface